### ISSUE TITLE
Reproducible crash when filter would exclude all items one could grab from vehicle storage

### ISSUE THREAD

**Issue by Svatopluk1** [CleverRaven/Cataclysm-DDA#50770]
### Describe the bug

You may have a lot of items stashed in a vehicle storage space. If you want to grab a specific item from a location with a lot of stuff, then, instead of pressing Page Down 17 times, it is more efficient to search (filter) for the thing you want. However, if it turns out that the filter would return no result (e.g. because the thing you searched for is not there, or you make a typo), the game crashes.

### Steps To Reproduce

1. Debug spawn an undamaged Cube Van, and some apples.
2. Walk to the cargo area, drop some apples into its Cargo Space.
3. Examine the Cargo Space you dropped the apples into, select the Get Items menu.
4. Filter (press F or / by default), set the filter to something that does not exist in the storage space, like Orange.
5. Crash as soon as you set the filter.

### Expected behavior

Show no result, or a warning message that the filter matches no items at that location.
Not crash to desktop.

### Versions and configuration

- OS: Linux
    - OS Version: LSB Version: 1.4; Distributor ID: Arch; Description: Arch Linux; Release: rolling; Codename: n/a; 
- Game Version: 0.F [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions]
]

Custom-built from the latest git master via Arch AUR: **cataclysm-dda-git 0.F.2021.08.14-1**

### Additional context

The program has crashed.
See the log file for a stack trace.
VERSION: 0.F
TYPE: Signal
MESSAGE: SIGABRT: Abnormal termination
STACK TRACE:

    cataclysm-tiles(debug_write_backtrace(std::ostream&)+0x46) [0x564e0fe9d7b9]
    cataclysm-tiles(+0x8e6152) [0x564e0fe7e152]
    cataclysm-tiles(+0x8e63db) [0x564e0fe7e3db]
    /usr/lib/libc.so.6(+0x3cda0) [0x7f2c18305da0]
    /usr/lib/libc.so.6(gsignal+0x142) [0x7f2c18305d22]
    /usr/lib/libc.so.6(abort+0x116) [0x7f2c182ef862]
    cataclysm-tiles(+0x65496b) [0x564e0fbec96b]
    cataclysm-tiles(std::vector<int, std::allocator<int> >::operator[](unsigned long)+0x37) [0x564e0fc535b5]
    cataclysm-tiles(+0x1067531) [0x564e105ff531]
    cataclysm-tiles(ui_adaptor::redraw_invalidated()+0x174) [0x564e107c1462]
    cataclysm-tiles(query_popup::query_once()+0xd5) [0x564e106371e3]
    cataclysm-tiles(query_popup::query()+0x48) [0x564e106377c6]
    cataclysm-tiles(popup(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, PopupFlags)+0xd6) [0x564e10574fa5]
    cataclysm-tiles(Pickup::pick_up(tripoint const&, int, Pickup::from_where)+0x223a) [0x564e106031b8]
    cataclysm-tiles(vehicle::interact_with(vpart_position const&)+0x2355) [0x564e10869393]
    cataclysm-tiles(game::examine(tripoint const&)+0x441) [0x564e0ffe9745]
    cataclysm-tiles(game::examine()+0x1ce) [0x564e0ffea056]
    cataclysm-tiles(game::do_regular_action(action_id&, avatar&, cata::optional<tripoint> const&)+0x1073) [0x564e1003557b]
    cataclysm-tiles(game::handle_action()+0xb9f) [0x564e1003abf3]
    cataclysm-tiles(do_turn()+0x542) [0x564e0fee911a]

    Attempting to repeat stack trace using debug symbols…
    debug_write_backtrace(std::ostream&)
    ??:?
    init_crash_handlers()
    ??:?
    init_crash_handlers()
    ??:?
    __restore_rt
    sigaction.c:?
    __GI_raise
    :?
    __GI_abort
    :?
    std::__replacement_assert(char const*, int, char const*, char const*)
    ??:?
    std::vector<int, std::allocator<int> >::operator[](unsigned long)
    ??:?
    Pickup::query_thief()
    ??:?
    ui_adaptor::redraw_invalidated()
    ??:?
    query_popup::query_once()
    ??:?
    query_popup::query()
    ??:?
    popup(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, PopupFlags)
    ??:?
    Pickup::pick_up(tripoint const&, int, Pickup::from_where)
    ??:?
    vehicle::interact_with(vpart_position const&)
    ??:?
    game::examine(tripoint const&)
    ??:?
    game::examine()
    ??:?
    game::do_regular_action(action_id&, avatar&, cata::optional<tripoint> const&)
    ??:?
    game::handle_action()
    ??:?
    do_turn()
    ??:?

**Comment by andrei8l** [comment]
Looks like a duplicate of https://github.com/CleverRaven/Cataclysm-DDA/issues/48858

**Comment by Svatopluk1** [comment]
Right, sorry. I did do a search for crash bugs, must have overlooked it.

**Comment by Saicchi** [comment]
I still can not reproduce it on experimental.
- OS: Linux
    - OS Version: LSB Version: 1.4; Distributor ID: Arch; Description: Arch Linux; Release: rolling; Codename: n/a; 
- Game Version: 0.F-1687-gbadd5e3393 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]

**Comment by Svatopluk1** [comment]
I don't have `-fsanitize=address` or `-fsanitize=address,undefined` explicitly specified (I did a text search for `fsanitize` just to be sure).

The only difference from a "standard" Arch config I can think of is that I like to build with `-march=native`. However, I have now rebuilt the latest master with the default `-march=x86-64 -mtune=generic`, and the game still crashes reproducibly using the steps above.

In the code snippet andrei8l highlighted in #48858, it looks like the current version dereferences the first item from a list that may be empty (if the player specifies a filter that matches nothing). Surely that is a potential source for errors, and not good practice in general, even if some compilers/settings sidestep the issue?
